### ISSUE TITLE
Define workspace settings for consistent formatting for eol and tab sizes and migrate to eslint

### ISSUE THREAD

**Issue by ItaloBorrelli** [jdneo/vscode-checkstyle#366]
If it's ok, I'd like to suggest the addition of whatever the maintainers have for their settings for formatting. I'm not sure if these are all correct or accurate to what other maintainers and developers have been using on this project, so please feel free to offer suggestions. I work on a few different projects on my machine with vscode and it'd be helpful if there were project specific settings for this project. Alternatively, we could `.gitignore` `settings.json` and offer suggested settings for our contribution guide. I might suggest the following:

```
{
  "files.eol": "\n",
  "[java]": {
    "editor.tabSize": 4,
    "editor.defaultFormatter": "redhat.java",
  },
  "[json]": {
    "editor.tabSize": 2,
    "editor.defaultFormatter": "esbenp.prettier-vscode",
  },
  "[ts]": {
    "editor.tabSize": 2,
    "editor.defaultFormatter": "esbenp.prettier-vscode",
  },
  "[xml]": {
    "editor.tabSize": 2,
  }
}
```

Ok, so, regarding the tab size, I can see that all the tabs in this project are 4 spaces, but 2 space tabs is consistent with most front end development I've seen.

Also, perhaps we could include the tab size, whether 2 or 4, in linting. Lastly in this, it might be a good opportunity to migrate to eslint.

Let me know if you have any other suggestions for this.

**Comment by ItaloBorrelli** [comment]
Also, it seems most files have \n for eol, but there's some that have \r:
```
❯ git ls-files | xargs grep -l $'\r'
.gitignore
.vscode/extensions.json
.vscode/tasks.json
CONTRIBUTING.md
README.md
docs/gifs/demo.gif
docs/gifs/liveLinting.gif
docs/gifs/setConfiguration.gif
docs/gifs/setVersion(lower).gif
docs/imgs/btn_bulb.png
docs/imgs/quick_fix.png
docs/imgs/set_config.png
docs/imgs/set_version.png
jdtls.ext/.mvn/wrapper/maven-wrapper.jar
jdtls.ext/com.shengchen.checkstyle.checker/pom.xml
jdtls.ext/com.shengchen.checkstyle.runner/.classpath
jdtls.ext/com.shengchen.checkstyle.runner/.project
jdtls.ext/com.shengchen.checkstyle.runner/.settings/org.eclipse.core.resources.prefs
jdtls.ext/com.shengchen.checkstyle.runner/.settings/org.eclipse.m2e.core.prefs
jdtls.ext/com.shengchen.checkstyle.runner/META-INF/MANIFEST.MF
jdtls.ext/com.shengchen.checkstyle.target/target.target
resources/icon_checkstyle.png
```

**Comment by jdneo** [comment]
I have no strong opinion on the code styles. As long as it makes the project setup easier, it's worth to do it. Thank you for the contribution and please go ahead.

**Comment by jdneo** [comment]
Addressed by #367
